cmd/rollkit/commands/toml.go

```diff
@@ -49,19 +50,69 @@ var initCmd = &cobra.Command{
 			fmt.Printf("Found rollup configuration under %s, adding to rollkit.toml\n", chainConfigDir)
 		}
 
-		config := rollconf.TomlConfig{
-			Entrypoint: entrypoint,
-			Chain: rollconf.ChainTomlConfig{
-				ConfigDir: chainConfigDir,
-			},
+		// Create a config with default values
+		config := rollconf.DefaultNodeConfig
+
+		// Update with the values we found
+		config.Entrypoint = entrypoint
+		config.Chain.ConfigDir = chainConfigDir
+
+		// Set the root directory to the current directory
+		currentDir, err := os.Getwd()
+		if err != nil {
+			fmt.Println("Error getting current directory:", err)
+			os.Exit(1)
+		}
+		config.RootDir = currentDir
+
+		// Create a new Viper instance to avoid conflicts with any existing configuration
+		v := viper.New()
+
+		// Configure Viper to use the structure directly
+		v.SetConfigName(rollconf.ConfigBaseName)
+		v.SetConfigType(rollconf.ConfigExtension)
+		v.AddConfigPath(currentDir)
+
+		// Create a map with the configuration structure
```

Comment:
Adding the fields manually is viper is a bit weird.
Normally, `AddConfigPath`, `BindPFlag`, `BindEnv`, `ReadInConfig` should be sufficient.
The keys would be the toml key path, matching what this is doing.
Every new field addition would need to be added there otherwise.